libs/src/services/solanaWeb3Manager/transfer.js

```diff
@@ -0,0 +1,153 @@
+const {
+  PublicKey,
+  SYSVAR_INSTRUCTIONS_PUBKEY
+} = require('@solana/web3.js')
+const BN = require('bn.js')
+const borsh = require('borsh')
+const keccak256 = require('keccak256')
+const secp256k1 = require('secp256k1')
+
+/**
+ * Transfer wAUDIO between wallets on solana
+ */
+
+class TransferInstructionData {
+  constructor ({
+    ethAddress,
+    amount
+  }) {
+    this.eth_address = ethAddress
+    this.amount = amount
+  }
+}
+
+const transferInstructionSchema = new Map([
+  [
+    TransferInstructionData,
+    {
+      kind: 'struct',
+      fields: [
+        ['eth_address', [20]],
+        ['amount', 'u64']
+      ]
+    }
+  ]
+])
+
+// transferWAudioBalance transfers wrapped Audio from one generated solana account to another.
+// For it to work, you have to have the eth private key belonging to the eth public key
+// that generated the solana account
+async function transferWAudioBalance (
+  amount,
```

Comment:
would be much nicer to have an object w labeled args rather than positional args here imo